syntax = "proto3";
package task;




import "google/protobuf/struct.proto";

/*
UpdateTaskDetail请求
*/
message UpdateTaskDetailRequest {
message Annotations {
    /*
    应用ID
    */
    string appId = 1 ;
}
        
message CmdConfig {
    /*
    uri
    */
    string url = 1 ;
    /*
    名字服务
    */
    string service_name = 2 ;
    /*
    端口
    */
    int32 port = 3 ;
    /*
    名字服务
    */
    string host = 4 ;
    /*
    http 请求
    */
    string method = 5 ;
    /*
    请求头
    */
    google.protobuf.Struct headers = 6 ;
    /*
    参数
    */
    google.protobuf.Struct params = 7 ;
}
        
message Callback {
    /*
    url
    */
    string url = 1 ;
    /*
    header host
    */
    string host = 2 ;
    /*
    名字服务
    */
    string ensName = 3 ;
}
        
    
    /*
    任务id
    */
    string taskId = 1 ;
    /*
    0 在页面可见，1 不可见
    */
    int32 invisible = 2 ;
    /*
    0 表示不禁用, 1 表示禁用
    */
    int32 disable = 3 ;
    /*
    运行状态
    */
    int32 status = 4 ;
    /*
    调度结果信息。{u'message': u'Success', u'code': 0, u'data': {u'execId': u'201809_task155123648b640b916475'}, u'error': u'\u6210\u529f\u3002Success'}
    */
    string error = 5 ;
    /*
    任务对象类型
    */
    string job_type = 6 ;
    /*
    相关参数
    */
    Annotations annotations = 7 ;
    /*
    定时任务执行周期类型
    */
    string task_type = 8 ;
    /*
    定时任务名称
    */
    string name = 9 ;
    /*
    任务的具体配置, http或脚本
    */
    CmdConfig cmd_config = 10 ;
    /*
    定时时间：x/5 x x x x， 2018-09-04 15:56:23
    */
    string task_scheduler = 11 ;
    /*
    回调
    */
    Callback callback = 12 ;
    /*
    更新权限
    */
    repeated string updateAuthorizers = 13 ;
    /*
    操作权限
    */
    repeated string operateAuthorizers = 14 ;
    /*
    删除权限
    */
    repeated string deleteAuthorizers = 15 ;
}

/*
UpdateTaskDetail返回
*/
message UpdateTaskDetailResponse {
message Data {
    /*
    修改成功的数目
    */
    int32 count = 1 ;
}
        
    
    /*
    返回码
    */
    int32 code = 1 ;
    /*
    结果信息
    */
    string msg = 2 ;
    /*
    返回数据
    */
    Data data = 3 ;
}

/*
UpdateTaskDetailApi返回
*/
message UpdateTaskDetailResponseWrapper {
    /*
    返回码
    */
    int32 code = 1 ;
    /*
    返回码解释
    */
    string codeExplain = 2 ;
    /*
    错误详情
    */
    string error = 3 ;
    /*
    返回数据
    */
    task.UpdateTaskDetailResponse data = 4 ;
}

